src/scripts/main.js

```diff
@@ -1 +1,75 @@
 'use strict';
+
+const promise1 = new Promise((resolve, reject) => {
+  let isClick = false;
+  const onClick = (e) => {
+    if (e.button === 0) {
+      isClick = true;
+      resolve('First promise was resolved');
+      document.removeEventListener('click', onClick);
+    }
+  };
+
+  document.addEventListener('click', onClick);
+
+  setTimeout(() => {
+    if (!isClick) {
+      reject(new Error('First promise was rejected'));
+      document.removeEventListener('click', onClick);
+    }
+  }, 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  const onClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+      document.removeEventListener('click', onClick);
+    }
+  };
+
+  document.addEventListener('click', onClick);
+});
+
+const promise3 = new Promise((resolve) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      leftClick = true;
+    }
+
+    if (e.button === 2) {
+      rightClick = true;
+    }
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
```

Comment:
The third promise does not remove the event listener after resolving. This could lead to memory leaks or unexpected behavior if the event listener remains active. Consider removing the event listener once the promise is resolved, similar to how it's done in the first two promises.